core/src/Repository/DataSource/Sql/Helper/SqlBuilder.php

```diff
@@ -138,12 +134,47 @@ public function multiInsert(array $entities): Query {
     return $query;
   }
 
-  public function deleteById(mixed $value): Query {
-    $query = $this->factory
-      ->delete($this->schema->getTableName())
-      ->where(field($this->schema->getIdColumn())->eq($value))
-      ->compile();
+  public function deleteById(mixed $value): SqlQuery {
+    if ($this->schema->softDeleteEnabled()) {
+      $now = date("Y-m-d H:i:s");
+      $factory = $this->factory
+        ->update($this->schema->getTableName(), [
+          SqlBaseColumn::DELETED_AT => $now,
+        ])
+        ->where(field($this->schema->getIdColumn())->eq($value));
+    } else {
+      $factory = $this->factory
+        ->delete($this->schema->getTableName())
+        ->where(field($this->schema->getIdColumn())->eq($value));
+    }
+
+    $query = $factory->compile();
 
     return $query;
   }
+
+  /**
+   * @param Query $query
+   * @param SelectQuery|UpdateQuery $factory
+   *
+   * @return SelectQuery|UpdateQuery
+   */
+  public function addWhereConditions(
+    Query $query,
+    SelectQuery|UpdateQuery $factory
+  ): SelectQuery|UpdateQuery {
+    if ($query instanceof WhereQuery) {
+      $wheres = $query->where();
+
+      foreach ($wheres as $column => $value) {
+        $factory->andWhere(field($column)->eq($value));
+      }
+    }
+
+    if ($this->schema->softDeleteEnabled()) {
```

Comment:
@orioljp what do you think if we add a third parameter to use or not the softdelete in this case?.
For example, if we want to SELECT all rows that was soft deleted, now is not possible.

Ex:
```
public function addWhereConditions(
    Query $query,
    SelectQuery|UpdateQuery $factory,
    bool $filterSoftDelete = true,
  ): SelectQuery|UpdateQuery {
...
if ($this->schema->softDeleteEnabled() && $filterSoftDelete) {
...
```

Reply:
Note: already discussed that we will add it in Query Composition.